### ISSUE TITLE
Ch4 Fix solution for original factorizations exercise

### ISSUE THREAD

**Issue by milesfrain** [purescript-contrib/purescript-book#208]
This involves reverting most of #207 and then fixing the exercise.
The current state is better than it was before though, since at least there's no mismatch between the exercise text and solution.

**Comment by klarkc** [comment]
I would like to add my solution here, it was very simple and involves the other functions already made:

```purs
factorize :: Int -> Array Int
factorize n = do
  f <- reverse $ factors n
  fi <- reverse $ f
  guard $ isPrime fi
  pure fi
```
[source](https://github.com/klarkc/purescript-book/blob/4a15bc8596bcb0cf1069d598b4c5f53a16ad1e75/exercises/chapter4/test/MySolutions.purs#L64-L69)

**Comment by milesfrain** [comment]
Fixed by #387
